sdc/functions/numpy_like.py

```diff
@@ -97,3 +97,30 @@ def sdc_astype_number_impl(self, dtype):
             return arr
 
         return sdc_astype_number_impl
+
+
+def nan_min_max_factory(reduce_op):
+    def impl(a):
+        length = len(a)
+
+        # set first element
+        result = None
+        for i in range(length):
```

Comment:
I still don't like this solution.

1. It is not scalable
2. It most likely would prevent loops fusing
3. It is big and ugly
4. It is unnecessary for ints

Reply:
For max case you could set initial value to min_int, min_float. It is one line and you wouldn't need to do this loop. Though, in case of floats you need to somehow detect no value case